packages/frontend-v2/components/Plan/PlanDropdown.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { Dispatch, SetStateAction } from "react";
+import { Select } from "@chakra-ui/react";
+import { PlanModel } from "@graduate/common";
+
+interface PlanDropdownProps {
+  setSelectedPlanId: Dispatch<SetStateAction<number | undefined | null>>;
+  plans: PlanModel<string>[];
+  selectedPlanId: number | undefined | null;
+}
+export const PlanDropdown: React.FC<PlanDropdownProps> = ({
+  setSelectedPlanId,
+  plans,
+  selectedPlanId,
+}) => {
+  return (
+    <Select
+      placeholder="Select Plan"
+      width="15%"
+      mb="sm"
+      borderRadius="0"
+      value={selectedPlanId ? selectedPlanId : undefined}
+      onChange={(e) => {
+        if (!e.target.value) {
+          // no plan is selected, indicated using null(different from undef which is the initial state)
+          setSelectedPlanId(null);
```

Comment:
is there a reason why we don't use undef instead of null here? what is the importance of differentiating between unselected, vs no plan. Idk if its good to have a situation where there is just no plan in the home page

Reply:
You're right! The unselected is when the user selects the "no selection" option in the dropdown. Now it makes sense to always display a plan but dropdown menus normally allow you to have the "no selection" option. Shall discuss this with designers.